packages/web3-errors/src/errors/transaction_errors.ts

```diff
@@ -84,25 +85,54 @@ export class RevertInstructionError extends BaseWeb3Error {
 	}
 }
 
-export class TransactionRevertError extends BaseWeb3Error {
+export class TransactionRevertError<ReceiptType = TransactionReceipt> extends BaseWeb3Error {
 	public code = ERR_TX_REVERT_TRANSACTION;
 
 	public constructor(
 		public reason: string,
 		public signature?: string,
-		public receipt?: TransactionReceipt,
+		public receipt?: ReceiptType,
+		public data?: string,
 	) {
-		super(
-			`Transaction has been reverted by the EVM:\n ${JSON.stringify(receipt, undefined, 2)}`,
-		);
+		super(reason);
+	}
+
+	public toJSON() {
+		return {
+			...super.toJSON(),
+			reason: this.reason,
+			signature: this.signature,
+			receipt: this.receipt,
+			data: this.data
+		};
+	}
+}
+
+export class TransactionRevertWithCustomError<ReceiptType = TransactionReceipt> extends TransactionRevertError<ReceiptType> {
```

Comment:
This error is specifically for transactions that reverted when calling a contract method that had a custom error. I believe `web3-eth-contract` already has the logic, but this adds support for `web3Eth.sendTransaction`